IntegrationTest/AuthenticationEndpoints.cs

```diff
@@ -0,0 +1,160 @@
+
+		var response = await _client.PostAsJsonAsync("/refresh", dto);
+		
+		
+		Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);	
+	}
+	
+	[Fact]
+	public async Task RefreshToken_ShouldReturn_BadRequest_UserDoesntExist()
+	{
+		using var scope = _factory.Services.CreateScope();
+		var tokenRepo = scope.ServiceProvider.GetService<ITokenRepository>();
+		var userRepo = scope.ServiceProvider.GetService<IUserRepository>();
+		var rf = new RefreshToken{UserId = 1};
+		tokenRepo!.GetAccessTokenByRefreshTokenAsync(Arg.Any<string>()).Returns(Result.Ok(rf));
+		userRepo!.GetRolesByUserIdAsync(Arg.Any<int>()).Returns(new List<Role>());
+		var dto = new RefreshDto("token");
+
+		var response = await _client.PostAsJsonAsync("/refresh", dto);
+		
+		
+		Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);	
+	}
+	[Fact]
+	public async Task RefreshToken_ShouldReturn_OkLoginResponse()
+	{
+		using var scope = _factory.Services.CreateScope();
+		var tokenRepo = scope.ServiceProvider.GetService<ITokenRepository>();
+		var userRepo = scope.ServiceProvider.GetService<IUserRepository>();
+		var rf = new RefreshToken{UserId = 1};
+		tokenRepo!.GetAccessTokenByRefreshTokenAsync(Arg.Any<string>()).Returns(Result.Ok(rf));
+		userRepo!.GetRolesByUserIdAsync(Arg.Any<int>()).Returns(new List<Role>{new Role{RoleName = "Instructor"}});
+		var dto = new RefreshDto("token");
+
+		var response = await _client.PostAsJsonAsync("/refresh", dto);
+		
+		
+		Assert.True(response.IsSuccessStatusCode);
+	}
+	
+	// login
+	[Fact]
+	public async Task Login_ShouldReturn_WrongPassword()
+	{
+		using var scope = _factory.Services.CreateScope();
+		var userRepo = scope.ServiceProvider.GetService<IUserRepository>();
+		var x = new PasswordHasher<User>();
+		var user = new User { Id = 1, PasswordHash = "password", Email = "email@email.com", Name = "name", CreatedAt = DateTime.UtcNow};
+		var hash = x.HashPassword(user, "pass1");
+		user.PasswordHash = hash;
+		userRepo!.GetRolesByUserIdAsync(Arg.Any<int>()).Returns(new List<Role>{new Role{RoleName = "Instructor"}});
+		userRepo!.GetUserByEmailAsync(Arg.Any<string>()).Returns(user);
+		var dto = new LoginDto("email", "pass");
+		
+		var response = await _client.PostAsJsonAsync("/login", dto);
+		
+		Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);
+	}
+	[Fact]
+	public async Task Login_ShouldReturn_OkLoginResponse()
+	{
+		using var scope = _factory.Services.CreateScope();
+		var userRepo = scope.ServiceProvider.GetService<IUserRepository>();
+		var x = new PasswordHasher<User>();
+		var user = new User { Id = 1, PasswordHash = "password", Email = "email@email.com", Name = "name", CreatedAt = DateTime.UtcNow};
+		var hash = x.HashPassword(user, "pass");
+		user.PasswordHash = hash;
+		userRepo!.GetRolesByUserIdAsync(Arg.Any<int>()).Returns(new List<Role>{new Role{RoleName = "Instructor"}});
+		userRepo!.GetUserByEmailAsync(Arg.Any<string>()).Returns(user);
+		var dto = new LoginDto("email", "pass");
```

Comment:
I would expect `"email"` having to match the `"email@emil.com"` which the user is registered on above.

What am I not understanding fully?